// @workday/canvas-kit-react-button/dist/commonjs/lib/ButtonBase.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.ButtonLabelIcon = exports.ButtonLabelData = exports.ButtonBaseLabel = exports.ButtonBaseCon = undefined;

var _react = require('react');

var React = _interopRequireWildcard(_react);

var _canvasKitLabsReactCore = require('@workday/canvas-kit-labs-react-core');

var _isPropValid = require('@emotion/is-prop-valid');

var _isPropValid2 = _interopRequireDefault(_isPropValid);

var _types = require('./types');

var _canvasKitReactIcon = require('@workday/canvas-kit-react-icon');

var _ButtonStyles = require('./ButtonStyles');

var ButtonStyles = _interopRequireWildcard(_ButtonStyles);

var _utils = require('./utils');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

var __extends = undefined && undefined.__extends || function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf || { __proto__: [] } instanceof Array && function (d, b) {
            d.__proto__ = b;
        } || function (d, b) {
            for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
        };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() {
            this.constructor = d;
        }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
}();
var __assign = undefined && undefined.__assign || function () {
    __assign = Object.assign || function (t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = undefined && undefined.__rest || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0) t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function") for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
        if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i])) t[p[i]] = s[p[i]];
    }
    return t;
};
var ButtonBaseCon = exports.ButtonBaseCon = (0, _canvasKitLabsReactCore.styled)('button', {
    shouldForwardProp: function (prop) {
        return (0, _isPropValid2.default)(prop) && prop !== 'size';
    }
})(function (_a) {
    var variant = _a.variant,
        size = _a.size;
    if (variant === undefined) {
        return {};
    }
    var baseButton = (0, _utils.getBaseButton)(variant);
    var buttonStyles = (0, _utils.getButtonStyle)(baseButton, variant);
    var sizeStyles = size !== undefined ? (0, _utils.getButtonSize)(baseButton, size) : {};
    return __assign({}, baseButton.styles, buttonStyles, sizeStyles);
}, function (_a) {
    var grow = _a.grow;
    return grow && { width: '100%', maxWidth: '100%' };
});
var ButtonBaseLabel = exports.ButtonBaseLabel = (0, _canvasKitLabsReactCore.styled)('span', {
    shouldForwardProp: function (prop) {
        return (0, _isPropValid2.default)(prop) && prop !== 'size';
    }
})(ButtonStyles.labelBaseStyles.styles, function (_a) {
    var size = _a.size;
    var sizes = ButtonStyles.labelBaseStyles.variants.sizes;
    switch (size) {
        case _types.ButtonSize.Large:
        default:
            return sizes.large;
        case _types.ButtonSize.Small:
            return sizes.small;
        case _types.ButtonSize.Medium:
            return sizes.medium;
    }
}, function (_a) {
    var variant = _a.variant;
    var types = ButtonStyles.labelBaseStyles.variants.types;
    switch (variant) {
        case _types.TextButtonVariant.Default:
        case _types.TextButtonVariant.Inverse:
            return types.text;
        case _types.TextButtonVariant.AllCaps:
        case _types.TextButtonVariant.InverseAllCaps:
            return types.textAllCaps;
        case _types.DeprecatedButtonVariant.Primary:
            return types.deprecatedPrimary;
        case _types.DeprecatedButtonVariant.Secondary:
            return types.deprecatedSecondary;
        case _types.DeprecatedButtonVariant.Delete:
            return types.deprecatedDelete;
        default:
            return {};
    }
});
var ButtonLabelData = exports.ButtonLabelData = (0, _canvasKitLabsReactCore.styled)('span', {
    shouldForwardProp: function (prop) {
        return (0, _isPropValid2.default)(prop) && prop !== 'size';
    }
})(ButtonStyles.labelDataBaseStyles.styles, function (_a) {
    var size = _a.size;
    var sizes = ButtonStyles.labelDataBaseStyles.variants.sizes;
    switch (size) {
        case _types.ButtonSize.Large:
        default:
            return sizes.large;
        case _types.ButtonSize.Medium:
            return sizes.medium;
    }
});
var ButtonLabelIconStyled = (0, _canvasKitLabsReactCore.styled)('span', {
    shouldForwardProp: function (prop) {
        return (0, _isPropValid2.default)(prop) && prop !== 'size';
    }
})(ButtonStyles.labelIconBaseStyles.styles, function (_a) {
    var size = _a.size,
        dropdown = _a.dropdown;
    if (dropdown) {
        switch (size) {
            case _types.ButtonSize.Large:
            default:
                return { padding: '0 8px 0 0' };
            case _types.ButtonSize.Medium:
                return { padding: '0 4px 0 0' };
        }
    }
    var sizes = ButtonStyles.labelIconBaseStyles.variants.sizes;
    switch (size) {
        case _types.ButtonSize.Large:
        default:
            return sizes.large;
        case _types.ButtonSize.Medium:
            return sizes.medium;
    }
}, function (_a) {
    var iconPosition = _a.iconPosition;
    if (iconPosition === undefined) {
        return {};
    }
    var types = ButtonStyles.labelIconBaseStyles.variants.types;
    switch (iconPosition) {
        case _types.IconPosition.Left:
        default:
            return types.iconPositionLeft;
        case _types.IconPosition.Right:
            return types.iconPositionRight;
    }
});
var ButtonLabelIcon = function (_super) {
    __extends(ButtonLabelIcon, _super);
    function ButtonLabelIcon() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    ButtonLabelIcon.prototype.render = function () {
        var _a = this.props,
            icon = _a.icon,
            size = _a.size,
            dropdown = _a.dropdown,
            iconPosition = _a.iconPosition,
            elemProps = __rest(_a, ["icon", "size", "dropdown", "iconPosition"]);
        if (icon === undefined) {
            return {};
        }
        var iconSize = 24;
        if (size === _types.ButtonSize.Small) {
            iconSize = 20;
        }
        return React.createElement(ButtonLabelIconStyled, __assign({ iconPosition: iconPosition, dropdown: dropdown, size: size }, elemProps), React.createElement(_canvasKitReactIcon.SystemIcon, { size: iconSize, icon: icon }));
    };
    return ButtonLabelIcon;
}(React.Component);
exports.ButtonLabelIcon = ButtonLabelIcon;
