// @workday/canvas-kit-react-button/dist/commonjs/lib/ButtonColors.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.ButtonColors = undefined;

var _canvasKitReactCore = require('@workday/canvas-kit-react-core');

var _canvasKitReactCore2 = _interopRequireDefault(_canvasKitReactCore);

var _types = require('./types');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var __assign = undefined && undefined.__assign || function () {
    __assign = Object.assign || function (t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var _a;
var ButtonColors = exports.ButtonColors = (_a = {}, _a[_types.DeprecatedButtonVariant.Primary] = _canvasKitReactCore2.default.buttonColors.primary, _a[_types.DeprecatedButtonVariant.Secondary] = _canvasKitReactCore2.default.buttonColors.secondary, _a[_types.DeprecatedButtonVariant.Delete] = __assign({}, _canvasKitReactCore2.default.buttonColors.delete, { focusBorder: _canvasKitReactCore2.default.colors.cinnamon500, activeBorder: _canvasKitReactCore2.default.colors.cinnamon500 }), _a[_types.ButtonVariant.Primary] = {
    background: _canvasKitReactCore2.default.colors.blueberry400,
    border: 'transparent',
    text: _canvasKitReactCore2.default.colors.frenchVanilla100,
    activeBackground: _canvasKitReactCore2.default.colors.blueberry600,
    activeBorder: 'transparent',
    activeText: _canvasKitReactCore2.default.colors.frenchVanilla100,
    disabledBackground: _canvasKitReactCore2.default.colors.blueberry200,
    disabledBorder: 'transparent',
    disabledText: _canvasKitReactCore2.default.colors.frenchVanilla100,
    focusBackground: _canvasKitReactCore2.default.colors.blueberry400,
    focusBorder: 'transparent',
    focusText: _canvasKitReactCore2.default.colors.frenchVanilla100,
    hoverBackground: _canvasKitReactCore2.default.colors.blueberry500,
    hoverBorder: 'transparent',
    hoverText: _canvasKitReactCore2.default.colors.frenchVanilla100,
    labelIcon: _canvasKitReactCore2.default.colors.frenchVanilla100,
    labelIconHover: _canvasKitReactCore2.default.colors.frenchVanilla100,
    labelIconActive: _canvasKitReactCore2.default.colors.frenchVanilla100,
    labelIconFocus: _canvasKitReactCore2.default.colors.frenchVanilla100,
    labelIconDisabled: _canvasKitReactCore2.default.colors.frenchVanilla100
}, _a[_types.ButtonVariant.Secondary] = {
    background: _canvasKitReactCore2.default.colors.soap200,
    border: 'transparent',
    text: _canvasKitReactCore2.default.colors.blackPepper400,
    activeBackground: _canvasKitReactCore2.default.colors.soap500,
    activeBorder: 'transparent',
    activeText: _canvasKitReactCore2.default.colors.blackPepper400,
    disabledBackground: _canvasKitReactCore2.default.colors.soap100,
    disabledBorder: 'transparent',
    disabledText: _canvasKitReactCore2.default.colors.licorice100,
    focusBackground: _canvasKitReactCore2.default.colors.soap200,
    focusBorder: 'transparent',
    focusText: _canvasKitReactCore2.default.colors.blackPepper400,
    hoverBackground: _canvasKitReactCore2.default.colors.soap400,
    hoverBorder: 'transparent',
    hoverText: _canvasKitReactCore2.default.colors.blackPepper400,
    labelIcon: _canvasKitReactCore2.default.colors.licorice200,
    labelIconHover: _canvasKitReactCore2.default.colors.licorice500,
    labelIconActive: _canvasKitReactCore2.default.colors.licorice500,
    labelIconFocus: _canvasKitReactCore2.default.colors.licorice200,
    labelIconDisabled: _canvasKitReactCore2.default.colors.soap600,
    labelData: _canvasKitReactCore2.default.colors.blackPepper400,
    labelDataDisabled: _canvasKitReactCore2.default.colors.licorice100
}, _a[_types.ButtonVariant.Delete] = {
    background: _canvasKitReactCore2.default.colors.cinnamon500,
    border: _canvasKitReactCore2.default.colors.cinnamon500,
    text: _canvasKitReactCore2.default.colors.frenchVanilla100,
    activeBackground: '#80160E',
    activeBorder: 'transparent',
    activeText: _canvasKitReactCore2.default.colors.frenchVanilla100,
    disabledBackground: _canvasKitReactCore2.default.colors.cinnamon200,
    disabledBorder: 'transparent',
    disabledText: _canvasKitReactCore2.default.colors.frenchVanilla100,
    focusBackground: _canvasKitReactCore2.default.colors.cinnamon500,
    focusText: _canvasKitReactCore2.default.colors.frenchVanilla100,
    hoverBackground: _canvasKitReactCore2.default.colors.cinnamon600,
    hoverBorder: _canvasKitReactCore2.default.colors.cinnamon600,
    hoverText: _canvasKitReactCore2.default.colors.frenchVanilla100
}, _a[_types.ButtonVariant.Highlight] = {
    background: _canvasKitReactCore2.default.colors.soap200,
    border: _canvasKitReactCore2.default.colors.soap200,
    text: _canvasKitReactCore2.default.colors.blueberry500,
    activeBackground: _canvasKitReactCore2.default.colors.soap500,
    activeBorder: 'transparent',
    activeText: _canvasKitReactCore2.default.colors.blueberry500,
    disabledBackground: _canvasKitReactCore2.default.colors.soap100,
    disabledBorder: 'transparent',
    disabledText: _canvasKitReactCore2.default.colors.licorice100,
    focusBackground: _canvasKitReactCore2.default.colors.soap200,
    focusBorder: 'transparent',
    focusText: _canvasKitReactCore2.default.colors.blueberry500,
    hoverBackground: _canvasKitReactCore2.default.colors.soap400,
    hoverBorder: 'transparent',
    hoverText: _canvasKitReactCore2.default.colors.blueberry500,
    labelIcon: _canvasKitReactCore2.default.colors.blueberry500,
    labelIconHover: _canvasKitReactCore2.default.colors.blueberry500,
    labelIconActive: _canvasKitReactCore2.default.colors.blueberry500,
    labelIconFocus: _canvasKitReactCore2.default.colors.blueberry500,
    labelIconDisabled: _canvasKitReactCore2.default.colors.soap600
}, _a[_types.ButtonVariant.OutlinePrimary] = {
    background: 'transparent',
    border: _canvasKitReactCore2.default.colors.blueberry400,
    text: _canvasKitReactCore2.default.colors.blueberry400,
    activeBackground: _canvasKitReactCore2.default.colors.blueberry500,
    activeBorder: 'transparent',
    activeText: _canvasKitReactCore2.default.colors.frenchVanilla100,
    disabledBackground: _canvasKitReactCore2.default.colors.frenchVanilla100,
    disabledBorder: _canvasKitReactCore2.default.colors.soap500,
    disabledText: _canvasKitReactCore2.default.colors.licorice100,
    focusBackground: _canvasKitReactCore2.default.colors.blueberry400,
    focusBorder: 'transparent',
    focusText: _canvasKitReactCore2.default.colors.frenchVanilla100,
    hoverBackground: _canvasKitReactCore2.default.colors.blueberry400,
    hoverBorder: 'transparent',
    hoverText: _canvasKitReactCore2.default.colors.frenchVanilla100,
    labelIcon: _canvasKitReactCore2.default.colors.blueberry400,
    labelIconHover: _canvasKitReactCore2.default.colors.frenchVanilla100,
    labelIconActive: _canvasKitReactCore2.default.colors.frenchVanilla100,
    labelIconFocus: _canvasKitReactCore2.default.colors.frenchVanilla100,
    labelIconDisabled: _canvasKitReactCore2.default.colors.soap600
}, _a[_types.ButtonVariant.OutlineSecondary] = {
    background: 'transparent',
    border: _canvasKitReactCore2.default.colors.soap500,
    text: _canvasKitReactCore2.default.colors.blackPepper400,
    activeBackground: _canvasKitReactCore2.default.colors.licorice600,
    activeBorder: 'transparent',
    activeText: _canvasKitReactCore2.default.colors.frenchVanilla100,
    disabledBackground: _canvasKitReactCore2.default.colors.frenchVanilla100,
    disabledBorder: _canvasKitReactCore2.default.colors.soap500,
    disabledText: _canvasKitReactCore2.default.colors.licorice100,
    focusBackground: _canvasKitReactCore2.default.colors.licorice500,
    focusBorder: 'transparent',
    focusText: _canvasKitReactCore2.default.colors.frenchVanilla100,
    hoverBackground: _canvasKitReactCore2.default.colors.licorice500,
    hoverBorder: 'transparent',
    hoverText: _canvasKitReactCore2.default.colors.frenchVanilla100,
    labelIcon: _canvasKitReactCore2.default.colors.licorice200,
    labelIconHover: _canvasKitReactCore2.default.colors.frenchVanilla100,
    labelIconActive: _canvasKitReactCore2.default.colors.frenchVanilla100,
    labelIconFocus: _canvasKitReactCore2.default.colors.frenchVanilla100,
    labelIconDisabled: _canvasKitReactCore2.default.colors.soap600
}, _a[_types.ButtonVariant.OutlineInverse] = {
    background: 'transparent',
    border: _canvasKitReactCore2.default.colors.frenchVanilla100,
    text: _canvasKitReactCore2.default.colors.frenchVanilla100,
    activeBackground: _canvasKitReactCore2.default.colors.soap300,
    activeBorder: 'transparent',
    activeText: _canvasKitReactCore2.default.colors.blackPepper400,
    disabledBackground: 'transparent',
    disabledBorder: 'rgba(255, 255, 255, 0.75)',
    disabledText: 'rgba(255, 255, 255, 0.75)',
    focusBackground: _canvasKitReactCore2.default.colors.frenchVanilla100,
    focusBorder: 'transparent',
    focusRingInner: 'currentColor',
    focusRingOuter: _canvasKitReactCore2.default.colors.frenchVanilla100,
    focusText: _canvasKitReactCore2.default.colors.blackPepper400,
    hoverBackground: _canvasKitReactCore2.default.colors.frenchVanilla100,
    hoverBorder: 'transparent',
    hoverText: _canvasKitReactCore2.default.colors.blackPepper400,
    labelIcon: _canvasKitReactCore2.default.colors.frenchVanilla100,
    labelIconHover: _canvasKitReactCore2.default.colors.licorice500,
    labelIconActive: _canvasKitReactCore2.default.colors.licorice500,
    labelIconFocus: _canvasKitReactCore2.default.colors.licorice500,
    labelIconDisabled: 'rgba(255, 255, 255, 0.75)',
    labelDataHover: _canvasKitReactCore2.default.colors.licorice300,
    labelDataActive: _canvasKitReactCore2.default.colors.licorice300,
    labelDataFocus: _canvasKitReactCore2.default.colors.licorice300,
    labelDataDisabled: 'rgba(255, 255, 255, 0.75)'
}, _a[_types.TextButtonVariant.Default] = {
    background: 'transparent',
    border: 'transparent',
    text: _canvasKitReactCore2.default.colors.blueberry400,
    activeBackground: 'transparent',
    activeBorder: 'transparent',
    activeText: _canvasKitReactCore2.default.colors.blueberry500,
    disabledBackground: 'transparent',
    disabledBorder: 'transparent',
    disabledText: 'rgba(8, 117, 225, 0.5)',
    focusBackground: 'transparent',
    focusText: _canvasKitReactCore2.default.colors.blueberry400,
    hoverBorder: 'transparent',
    hoverText: _canvasKitReactCore2.default.colors.blueberry500,
    labelIcon: _canvasKitReactCore2.default.colors.blueberry400,
    labelIconHover: _canvasKitReactCore2.default.colors.blueberry500,
    labelIconActive: _canvasKitReactCore2.default.colors.blueberry500,
    labelIconFocus: _canvasKitReactCore2.default.colors.blueberry400,
    labelIconDisabled: 'rgba(8, 117, 225, 0.5)'
}, _a[_types.TextButtonVariant.Inverse] = {
    background: 'transparent',
    border: 'transparent',
    text: _canvasKitReactCore2.default.colors.frenchVanilla100,
    activeBackground: 'transparent',
    activeBorder: 'transparent',
    activeText: _canvasKitReactCore2.default.colors.frenchVanilla100,
    disabledBackground: 'transparent',
    disabledBorder: 'transparent',
    disabledText: 'rgba(255, 255, 255, 0.5)',
    focusBackground: 'transparent',
    focusText: _canvasKitReactCore2.default.colors.frenchVanilla100,
    focusRingInner: 'transparent',
    focusRingOuter: _canvasKitReactCore2.default.colors.frenchVanilla100,
    hoverBorder: 'transparent',
    labelIcon: _canvasKitReactCore2.default.colors.frenchVanilla100,
    labelIconHover: 'currentColor',
    labelIconActive: _canvasKitReactCore2.default.colors.frenchVanilla100,
    labelIconFocus: _canvasKitReactCore2.default.colors.frenchVanilla100,
    labelIconDisabled: 'rgba(255, 255, 255, 0.5)'
}, _a[_types.TextButtonVariant.AllCaps] = null, _a[_types.TextButtonVariant.InverseAllCaps] = null, _a[_types.IconButtonVariant.Square] = {
    background: 'transparent',
    activeBackground: _canvasKitReactCore2.default.colors.soap500,
    disabledBackground: 'transparent',
    focusBackground: _canvasKitReactCore2.default.colors.soap300,
    hoverBackground: _canvasKitReactCore2.default.colors.soap300,
    labelIcon: _canvasKitReactCore2.default.colors.licorice200,
    labelIconHover: _canvasKitReactCore2.default.colors.licorice500,
    labelIconActive: _canvasKitReactCore2.default.colors.licorice500,
    labelIconFocus: _canvasKitReactCore2.default.colors.licorice500,
    labelIconFocusHover: _canvasKitReactCore2.default.colors.licorice500,
    labelIconDisabled: _canvasKitReactCore2.default.colors.soap600
}, _a[_types.IconButtonVariant.SquareFilled] = {
    background: _canvasKitReactCore2.default.colors.soap200,
    border: _canvasKitReactCore2.default.colors.soap500,
    activeBackground: _canvasKitReactCore2.default.colors.soap500,
    disabledBackground: _canvasKitReactCore2.default.colors.soap100,
    focusBackground: _canvasKitReactCore2.default.colors.soap400,
    hoverBackground: _canvasKitReactCore2.default.colors.soap400,
    labelIcon: _canvasKitReactCore2.default.colors.licorice200,
    labelIconHover: _canvasKitReactCore2.default.colors.licorice500,
    labelIconActive: _canvasKitReactCore2.default.colors.licorice500,
    labelIconFocus: _canvasKitReactCore2.default.colors.licorice500,
    labelIconDisabled: _canvasKitReactCore2.default.colors.soap600
}, _a[_types.IconButtonVariant.Plain] = {
    background: 'transparent',
    activeBackground: 'transparent',
    activeBorder: 'transparent',
    disabledBackground: 'transparent',
    focusBackground: 'transparent',
    hoverBackground: 'transparent',
    labelIcon: _canvasKitReactCore2.default.colors.licorice200,
    labelIconHover: _canvasKitReactCore2.default.colors.licorice500,
    labelIconActive: _canvasKitReactCore2.default.colors.licorice500,
    labelIconFocus: _canvasKitReactCore2.default.colors.licorice500,
    labelIconDisabled: _canvasKitReactCore2.default.colors.soap600
}, _a[_types.IconButtonVariant.Circle] = {
    background: 'transparent',
    activeBackground: _canvasKitReactCore2.default.colors.soap500,
    disabledBackground: 'transparent',
    focusBackground: _canvasKitReactCore2.default.colors.soap300,
    hoverBackground: _canvasKitReactCore2.default.colors.soap300,
    labelIcon: _canvasKitReactCore2.default.colors.licorice200,
    labelIconHover: _canvasKitReactCore2.default.colors.licorice500,
    labelIconActive: _canvasKitReactCore2.default.colors.licorice500,
    labelIconFocusHover: _canvasKitReactCore2.default.colors.licorice500,
    labelIconFocus: _canvasKitReactCore2.default.colors.licorice500,
    labelIconDisabled: _canvasKitReactCore2.default.colors.soap600
}, _a[_types.IconButtonVariant.CircleFilled] = {
    background: _canvasKitReactCore2.default.colors.soap200,
    activeBackground: _canvasKitReactCore2.default.colors.soap500,
    disabledBackground: _canvasKitReactCore2.default.colors.soap100,
    hoverBackground: _canvasKitReactCore2.default.colors.soap400,
    focusBackground: _canvasKitReactCore2.default.colors.soap400,
    labelIcon: _canvasKitReactCore2.default.colors.licorice200,
    labelIconHover: _canvasKitReactCore2.default.colors.licorice500,
    labelIconFocus: _canvasKitReactCore2.default.colors.licorice500,
    labelIconActive: _canvasKitReactCore2.default.colors.licorice500,
    labelIconDisabled: _canvasKitReactCore2.default.colors.soap600
}, _a[_types.IconButtonVariant.Inverse] = {
    background: 'transparent',
    activeBackground: 'rgba(0, 0, 0, 0.3)',
    focusBackground: 'rgba(0, 0, 0, 0.2)',
    disabledBackground: 'transparent',
    hoverBackground: 'rgba(0, 0, 0, 0.2)',
    labelIcon: _canvasKitReactCore2.default.colors.frenchVanilla100,
    labelIconHover: _canvasKitReactCore2.default.colors.frenchVanilla100,
    labelIconActive: _canvasKitReactCore2.default.colors.frenchVanilla100,
    labelIconFocus: _canvasKitReactCore2.default.colors.frenchVanilla100,
    labelIconDisabled: 'rgba(255, 255, 255, 0.75)',
    focusRingInner: 'currentColor',
    focusBorder: 'transparent',
    focusRingOuter: _canvasKitReactCore2.default.colors.frenchVanilla100,
    focusHover: 'rgba(0, 0, 0, 0.3)'
}, _a[_types.IconButtonVariant.InverseFilled] = {
    background: 'rgba(0, 0, 0, 0.2)',
    activeBackground: 'rgba(0, 0, 0, 0.4)',
    focusBackground: 'rgba(0, 0, 0, 0.2)',
    disabledBackground: 'rgba(0, 0, 0, 0.2)',
    hoverBackground: 'rgba(0, 0, 0, 0.3)',
    labelIcon: _canvasKitReactCore2.default.colors.frenchVanilla100,
    labelIconHover: _canvasKitReactCore2.default.colors.frenchVanilla100,
    labelIconActive: _canvasKitReactCore2.default.colors.frenchVanilla100,
    labelIconFocus: _canvasKitReactCore2.default.colors.frenchVanilla100,
    labelIconDisabled: 'rgba(255, 255, 255, 0.75)',
    focusRingInner: 'currentColor',
    focusBorder: 'transparent',
    focusRingOuter: _canvasKitReactCore2.default.colors.frenchVanilla100,
    focusHover: 'rgba(0, 0, 0, 0.3)'
}, _a);
exports.default = ButtonColors;
