// @workday/canvas-kit-react-button/dist/commonjs/lib/types.js
"use strict";

Object.defineProperty(exports, "__esModule", {
    value: true
});
var ButtonSize = exports.ButtonSize = undefined;
(function (ButtonSize) {
    ButtonSize["Small"] = "small";
    ButtonSize["Medium"] = "medium";
    ButtonSize["Large"] = "large";
})(ButtonSize || (exports.ButtonSize = ButtonSize = {}));
var IconButtonSize = exports.IconButtonSize = undefined;
(function (IconButtonSize) {
    IconButtonSize["Small"] = "small";
    IconButtonSize["Medium"] = "medium";
})(IconButtonSize || (exports.IconButtonSize = IconButtonSize = {}));
var DeprecatedButtonVariant = exports.DeprecatedButtonVariant = undefined;
(function (DeprecatedButtonVariant) {
    DeprecatedButtonVariant["Primary"] = "deprecatedPrimary";
    DeprecatedButtonVariant["Secondary"] = "deprecatedSecondary";
    DeprecatedButtonVariant["Delete"] = "deprecatedDelete";
})(DeprecatedButtonVariant || (exports.DeprecatedButtonVariant = DeprecatedButtonVariant = {}));
var ButtonVariant = exports.ButtonVariant = undefined;
(function (ButtonVariant) {
    ButtonVariant["Primary"] = "primary";
    ButtonVariant["Secondary"] = "betaSecondary";
    ButtonVariant["Delete"] = "delete";
    ButtonVariant["Highlight"] = "highlight";
    ButtonVariant["OutlinePrimary"] = "outlinePrimary";
    ButtonVariant["OutlineSecondary"] = "outlineSecondary";
    ButtonVariant["OutlineInverse"] = "outlineInverse";
})(ButtonVariant || (exports.ButtonVariant = ButtonVariant = {}));
var IconButtonVariant = exports.IconButtonVariant = undefined;
(function (IconButtonVariant) {
    IconButtonVariant["Square"] = "square";
    IconButtonVariant["SquareFilled"] = "squareFilled";
    IconButtonVariant["Plain"] = "plain";
    IconButtonVariant["Circle"] = "circle";
    IconButtonVariant["CircleFilled"] = "circleFilled";
    IconButtonVariant["Inverse"] = "inverse";
    IconButtonVariant["InverseFilled"] = "inverseFilled";
})(IconButtonVariant || (exports.IconButtonVariant = IconButtonVariant = {}));
var IconPosition = exports.IconPosition = undefined;
(function (IconPosition) {
    IconPosition["Left"] = "iconPositionLeft";
    IconPosition["Right"] = "iconPositionRight";
})(IconPosition || (exports.IconPosition = IconPosition = {}));
var TextButtonVariant = exports.TextButtonVariant = undefined;
(function (TextButtonVariant) {
    TextButtonVariant["Default"] = "text";
    TextButtonVariant["Inverse"] = "textInverse";
    TextButtonVariant["AllCaps"] = "textAllCaps";
    TextButtonVariant["InverseAllCaps"] = "textInverseAllCaps";
})(TextButtonVariant || (exports.TextButtonVariant = TextButtonVariant = {}));
