// @workday/canvas-kit-react-button/dist/commonjs/lib/utils.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.getButtonSize = getButtonSize;
exports.getButtonStyle = getButtonStyle;
exports.getBaseButton = getBaseButton;
exports.getButtonFocusRing = getButtonFocusRing;
exports.getButtonStateStyle = getButtonStateStyle;
exports.getIconButtonStateStyle = getIconButtonStateStyle;

var _canvasKitReactCommon = require('@workday/canvas-kit-react-common');

var _types = require('./types');

var _ButtonStyles = require('./ButtonStyles');

var ButtonStyles = _interopRequireWildcard(_ButtonStyles);

var _ButtonColors = require('./ButtonColors');

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

var __assign = undefined && undefined.__assign || function () {
    __assign = Object.assign || function (t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
function getButtonSize(baseButton, size) {
    var sizes = baseButton.variants.sizes;
    switch (size) {
        case _types.ButtonSize.Large:
            return sizes.large;
        case _types.ButtonSize.Medium:
        default:
            return sizes.medium;
        case _types.ButtonSize.Small:
            return sizes.small;
    }
}
function getButtonStyle(baseButton, variant) {
    var types = baseButton.variants.types;
    switch (variant) {
        case _types.DeprecatedButtonVariant.Primary:
            return types[_types.DeprecatedButtonVariant.Primary];
        case _types.DeprecatedButtonVariant.Secondary:
            return types[_types.DeprecatedButtonVariant.Secondary];
        case _types.DeprecatedButtonVariant.Delete:
            return types[_types.DeprecatedButtonVariant.Delete];
        case _types.ButtonVariant.Highlight:
            return types[_types.ButtonVariant.Highlight];
        case _types.ButtonVariant.OutlinePrimary:
            return types[_types.ButtonVariant.OutlinePrimary];
        case _types.ButtonVariant.OutlineSecondary:
            return types[_types.ButtonVariant.OutlineSecondary];
        case _types.ButtonVariant.OutlineInverse:
            return types[_types.ButtonVariant.OutlineInverse];
        case _types.ButtonVariant.Primary:
            return types[_types.ButtonVariant.Primary];
        case _types.ButtonVariant.Secondary:
        default:
            return types[_types.ButtonVariant.Secondary];
        case _types.ButtonVariant.Delete:
            return types[_types.ButtonVariant.Delete];
        case _types.TextButtonVariant.Default:
            return types[_types.TextButtonVariant.Default];
        case _types.TextButtonVariant.Inverse:
            return types[_types.TextButtonVariant.Inverse];
        case _types.TextButtonVariant.AllCaps:
            return types[_types.TextButtonVariant.AllCaps];
        case _types.TextButtonVariant.InverseAllCaps:
            return types[_types.TextButtonVariant.InverseAllCaps];
        case _types.IconButtonVariant.Square:
            return types[_types.IconButtonVariant.Square];
        case _types.IconButtonVariant.SquareFilled:
            return types[_types.IconButtonVariant.SquareFilled];
        case _types.IconButtonVariant.Plain:
            return types[_types.IconButtonVariant.Plain];
        case _types.IconButtonVariant.Circle:
            return types[_types.IconButtonVariant.Circle];
        case _types.IconButtonVariant.CircleFilled:
            return types[_types.IconButtonVariant.CircleFilled];
        case _types.IconButtonVariant.Inverse:
            return types[_types.IconButtonVariant.Inverse];
        case _types.IconButtonVariant.InverseFilled:
            return types[_types.IconButtonVariant.InverseFilled];
    }
}
function getBaseButton(buttonType) {
    switch (buttonType) {
        case _types.DeprecatedButtonVariant.Primary:
        case _types.DeprecatedButtonVariant.Secondary:
        case _types.DeprecatedButtonVariant.Delete:
            return ButtonStyles.deprecatedButtonStyles;
        case _types.ButtonVariant.Primary:
        case _types.ButtonVariant.Secondary:
        case _types.ButtonVariant.Delete:
        case _types.ButtonVariant.Highlight:
        case _types.ButtonVariant.OutlinePrimary:
        case _types.ButtonVariant.OutlineSecondary:
        case _types.ButtonVariant.OutlineInverse:
        default:
            return ButtonStyles.canvasButtonStyles;
    }
}
function getButtonFocusRing(variant) {
    var buttonColors = _ButtonColors.ButtonColors[variant];
    if (buttonColors == null) {
        return {};
    }
    switch (variant) {
        case _types.DeprecatedButtonVariant.Primary:
        case _types.DeprecatedButtonVariant.Secondary:
        case _types.TextButtonVariant.Default:
        case _types.TextButtonVariant.AllCaps:
            return (0, _canvasKitReactCommon.focusRing)(2, 0);
        case _types.ButtonVariant.OutlineInverse:
            return (0, _canvasKitReactCommon.focusRing)(2, 2, true, false, buttonColors.focusRingInner, buttonColors.focusRingOuter);
        case _types.IconButtonVariant.Plain:
            return (0, _canvasKitReactCommon.focusRing)(2);
        case _types.IconButtonVariant.Inverse:
        case _types.IconButtonVariant.InverseFilled:
            return (0, _canvasKitReactCommon.focusRing)(2, 2, true, false, buttonColors.focusRingInner, buttonColors.focusRingOuter);
        case _types.TextButtonVariant.Inverse:
        case _types.TextButtonVariant.InverseAllCaps:
            return (0, _canvasKitReactCommon.focusRing)(2, 0, true, false, buttonColors.focusRingInner, buttonColors.focusRingOuter);
        default:
            return (0, _canvasKitReactCommon.focusRing)(2, 2);
    }
}
function getButtonStateStyle(variant) {
    var _a, _b, _c, _d, _e;
    var buttonColors = _ButtonColors.ButtonColors[variant];
    if (buttonColors == null) {
        return {};
    }
    var baseStyles = __assign({ backgroundColor: buttonColors.background, borderColor: buttonColors.border, color: buttonColors.text }, buttonColors.labelIcon && {
        'span .wd-icon-fill, span .wd-icon-accent': {
            transition: 'fill 120ms ease-in',
            fill: buttonColors.labelIcon
        }
    }, buttonColors.labelData && (_a = {}, _a['.' + ButtonStyles.labelDataBaseStyles.classname] = {
        color: buttonColors.labelData
    }, _a));
    var hoverStyles = {
        ':hover': __assign({ backgroundColor: buttonColors.hoverBackground, borderColor: buttonColors.hoverBorder, color: buttonColors.hoverText }, buttonColors.labelDataHover && (_b = {}, _b['.' + ButtonStyles.labelDataBaseStyles.classname] = {
            transition: 'color 120ms ease-in',
            color: buttonColors.labelDataHover
        }, _b), buttonColors.labelIconHover && {
            'span .wd-icon-fill, span .wd-icon-accent': {
                fill: buttonColors.labelIconHover
            }
        })
    };
    var activeStyles = {
        ':active, :focus:active, :hover:active': __assign({ backgroundColor: buttonColors.activeBackground, borderColor: buttonColors.activeBorder, color: buttonColors.activeText }, buttonColors.labelDataActive && (_c = {}, _c['.' + ButtonStyles.labelDataBaseStyles.classname] = {
            color: buttonColors.labelDataActive
        }, _c), buttonColors.labelIconActive && {
            'span .wd-icon-fill, span .wd-icon-accent': {
                fill: buttonColors.labelIconActive
            }
        })
    };
    return __assign({}, baseStyles, { ':focus': __assign({ backgroundColor: buttonColors.focusBackground, borderColor: buttonColors.focusBorder, color: buttonColors.focusText }, buttonColors.labelDataFocus && (_d = {}, _d['.' + ButtonStyles.labelDataBaseStyles.classname] = {
            color: buttonColors.labelDataFocus
        }, _d), buttonColors.labelIconFocus && {
            'span .wd-icon-fill, span .wd-icon-accent': {
                fill: buttonColors.labelIconFocus
            }
        }) }, activeStyles, hoverStyles, { ':disabled, :active:disabled, :focus:disabled, :hover:disabled': __assign({ backgroundColor: buttonColors.disabledBackground, borderColor: buttonColors.disabledBorder, color: buttonColors.disabledText }, buttonColors.labelIconDisabled && {
            'span .wd-icon-fill, span .wd-icon-accent': {
                fill: buttonColors.labelIconDisabled
            }
        }, buttonColors.labelDataDisabled && (_e = {}, _e['.' + ButtonStyles.labelDataBaseStyles.classname] = {
            color: buttonColors.labelDataDisabled
        }, _e)), '&:not([disabled])': {
            '&:focus': __assign({ borderColor: buttonColors.focusBorder }, getButtonFocusRing(variant)),
            '&:active': __assign({ borderColor: buttonColors.activeBorder }, getButtonFocusRing(variant))
        } }, (0, _canvasKitReactCommon.mouseFocusBehavior)({
        '&:focus': __assign({}, baseStyles, { outline: 'none', boxShadow: 'none', animation: 'none' }, hoverStyles, activeStyles)
    }));
}
function getIconButtonStateStyle(variant) {
    var _a, _b, _c;
    var buttonColors = _ButtonColors.ButtonColors[variant];
    if (buttonColors == null) {
        return {};
    }
    var baseStyles = __assign({ borderColor: buttonColors.border, backgroundColor: buttonColors.background }, buttonColors.labelIcon && {
        'span .wd-icon-fill, span .wd-icon-accent': {
            transition: 'fill 120ms ease-in',
            fill: buttonColors.labelIcon
        }
    }, buttonColors.labelData && (_a = {}, _a['.' + ButtonStyles.labelDataBaseStyles.classname] = {
        color: buttonColors.labelData
    }, _a));
    var hoverStyles = {
        ':hover': __assign({ backgroundColor: buttonColors.hoverBackground }, buttonColors.labelIconHover && {
            'span .wd-icon-fill, span .wd-icon-accent': {
                fill: buttonColors.labelIconHover
            }
        })
    };
    var activeStyles = {
        ':active, :focus:active, :hover:active': __assign({ backgroundColor: buttonColors.activeBackground }, buttonColors.labelIconHover && {
            'span .wd-icon-fill, span .wd-icon-accent': {
                fill: buttonColors.labelIconActive
            }
        })
    };
    return __assign({}, baseStyles, { ':focus': __assign({ backgroundColor: buttonColors.focusBackground }, buttonColors.labelDataFocus && (_b = {}, _b['.' + ButtonStyles.labelDataBaseStyles.classname] = {
            color: buttonColors.labelDataFocus
        }, _b), buttonColors.labelIconFocus && {
            'span .wd-icon-fill, span .wd-icon-accent': {
                fill: buttonColors.labelIconFocus
            }
        }) }, hoverStyles, activeStyles, { ':disabled, :active:disabled, :focus:disabled, :hover:disabled': __assign({ pointerEvents: 'none', backgroundColor: buttonColors.disabledBackground, borderColor: buttonColors.disabledBorder, color: buttonColors.disabledText }, buttonColors.labelIconDisabled && {
            'span .wd-icon-fill, span .wd-icon-accent': {
                fill: buttonColors.labelIconDisabled
            }
        }, buttonColors.labelDataDisabled && (_c = {}, _c['.' + ButtonStyles.labelDataBaseStyles.classname] = {
            color: buttonColors.labelDataDisabled
        }, _c)), '&:not([disabled])': {
            '&:focus': __assign({ background: buttonColors.focusBackground, borderColor: buttonColors.focusBorder }, getButtonFocusRing(variant)),
            '&:active': __assign({ borderColor: buttonColors.activeBorder }, getButtonFocusRing(variant)),
            '&:hover:focus': {
                backgroundColor: buttonColors.focusHover,
                'span .wd-icon-fill, span .wd-icon-accent': {
                    fill: buttonColors.labelIconFocusHover
                }
            },
            '&:focus:active': {
                backgroundColor: buttonColors.activeBackground
            }
        } }, (0, _canvasKitReactCommon.mouseFocusBehavior)({
        '&:focus': __assign({}, baseStyles, { outline: 'none', boxShadow: 'none', animation: 'none' }, hoverStyles, activeStyles)
    }));
}
