// @workday/canvas-kit-react-button/dist/es6/lib/Button.js
var __extends = this && this.__extends || function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf || { __proto__: [] } instanceof Array && function (d, b) {
            d.__proto__ = b;
        } || function (d, b) {
            for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
        };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() {
            this.constructor = d;
        }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
}();
var __assign = this && this.__assign || function () {
    __assign = Object.assign || function (t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = this && this.__rest || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0) t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function") for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
        if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i])) t[p[i]] = s[p[i]];
    }
    return t;
};
import { jsx } from '@emotion/core';
import * as React from 'react';
import { ButtonBaseCon, ButtonBaseLabel, ButtonLabelData, ButtonLabelIcon } from './ButtonBase';
import { DeprecatedButtonVariant, ButtonSize, ButtonVariant } from './types';
import { labelDataBaseStyles } from './ButtonStyles';
var Button = function (_super) {
    __extends(Button, _super);
    function Button() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    Button.prototype.render = function () {
        var _a = this.props,
            variant = _a.variant,
            size = _a.size,
            buttonRef = _a.buttonRef,
            dataLabel = _a.dataLabel,
            icon = _a.icon,
            children = _a.children,
            elemProps = __rest(_a, ["variant", "size", "buttonRef", "dataLabel", "icon", "children"]);
        if (variant === ButtonVariant.Highlight && (icon === undefined || size === ButtonSize.Small)) {
            return null;
        }
        return jsx(ButtonBaseCon, __assign({ variant: variant, size: size, ref: buttonRef }, elemProps), icon && jsx(ButtonLabelIcon, { size: size, icon: icon }), jsx(ButtonBaseLabel, { size: size, variant: variant }, children), dataLabel && jsx(ButtonLabelData, { size: size, className: labelDataBaseStyles.classname }, dataLabel));
    };
    Button.Variant = ButtonVariant;
    Button.Size = ButtonSize;
    Button.defaultProps = {
        size: ButtonSize.Medium,
        variant: ButtonVariant.Secondary,
        grow: false
    };
    return Button;
}(React.Component);
export default Button;
var deprecated_Button = function (_super) {
    __extends(deprecated_Button, _super);
    function deprecated_Button() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    deprecated_Button.prototype.componentDidMount = function () {
        console.warn('This component is now deprecated, consider using the new Button component');
    };
    deprecated_Button.prototype.render = function () {
        var _a = this.props,
            variant = _a.variant,
            size = _a.size,
            buttonRef = _a.buttonRef,
            dataLabel = _a.dataLabel,
            icon = _a.icon,
            children = _a.children,
            elemProps = __rest(_a, ["variant", "size", "buttonRef", "dataLabel", "icon", "children"]);
        return jsx(Button, __assign({}, this.props, elemProps));
    };
    deprecated_Button.Variant = DeprecatedButtonVariant;
    deprecated_Button.Size = ButtonSize;
    deprecated_Button.defaultProps = {
        size: ButtonSize.Large,
        variant: DeprecatedButtonVariant.Secondary,
        grow: false
    };
    return deprecated_Button;
}(React.Component);
export { deprecated_Button };
