// @workday/canvas-kit-react-button/dist/es6/lib/types.js
export var ButtonSize;
(function (ButtonSize) {
    ButtonSize["Small"] = "small";
    ButtonSize["Medium"] = "medium";
    ButtonSize["Large"] = "large";
})(ButtonSize || (ButtonSize = {}));
export var IconButtonSize;
(function (IconButtonSize) {
    IconButtonSize["Small"] = "small";
    IconButtonSize["Medium"] = "medium";
})(IconButtonSize || (IconButtonSize = {}));
export var DeprecatedButtonVariant;
(function (DeprecatedButtonVariant) {
    DeprecatedButtonVariant["Primary"] = "deprecatedPrimary";
    DeprecatedButtonVariant["Secondary"] = "deprecatedSecondary";
    DeprecatedButtonVariant["Delete"] = "deprecatedDelete";
})(DeprecatedButtonVariant || (DeprecatedButtonVariant = {}));
export var ButtonVariant;
(function (ButtonVariant) {
    ButtonVariant["Primary"] = "primary";
    ButtonVariant["Secondary"] = "betaSecondary";
    ButtonVariant["Delete"] = "delete";
    ButtonVariant["Highlight"] = "highlight";
    ButtonVariant["OutlinePrimary"] = "outlinePrimary";
    ButtonVariant["OutlineSecondary"] = "outlineSecondary";
    ButtonVariant["OutlineInverse"] = "outlineInverse";
})(ButtonVariant || (ButtonVariant = {}));
export var IconButtonVariant;
(function (IconButtonVariant) {
    IconButtonVariant["Square"] = "square";
    IconButtonVariant["SquareFilled"] = "squareFilled";
    IconButtonVariant["Plain"] = "plain";
    IconButtonVariant["Circle"] = "circle";
    IconButtonVariant["CircleFilled"] = "circleFilled";
    IconButtonVariant["Inverse"] = "inverse";
    IconButtonVariant["InverseFilled"] = "inverseFilled";
})(IconButtonVariant || (IconButtonVariant = {}));
export var IconPosition;
(function (IconPosition) {
    IconPosition["Left"] = "iconPositionLeft";
    IconPosition["Right"] = "iconPositionRight";
})(IconPosition || (IconPosition = {}));
export var TextButtonVariant;
(function (TextButtonVariant) {
    TextButtonVariant["Default"] = "text";
    TextButtonVariant["Inverse"] = "textInverse";
    TextButtonVariant["AllCaps"] = "textAllCaps";
    TextButtonVariant["InverseAllCaps"] = "textInverseAllCaps";
})(TextButtonVariant || (TextButtonVariant = {}));
