// @workday/canvas-kit-react-button/dist/commonjs/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.IconButtonToggleGroup = exports.TextButton = exports.DropdownButton = exports.IconButton = exports.deprecated_Button = exports.beta_Button = exports.Button = undefined;

var _Button = require('./lib/Button');

Object.defineProperty(exports, 'Button', {
  enumerable: true,
  get: function () {
    return _interopRequireDefault(_Button).default;
  }
});
Object.defineProperty(exports, 'beta_Button', {
  enumerable: true,
  get: function () {
    return _interopRequireDefault(_Button).default;
  }
});
Object.defineProperty(exports, 'deprecated_Button', {
  enumerable: true,
  get: function () {
    return _Button.deprecated_Button;
  }
});

var _IconButton = require('./lib/IconButton');

Object.defineProperty(exports, 'IconButton', {
  enumerable: true,
  get: function () {
    return _interopRequireDefault(_IconButton).default;
  }
});

var _DropdownButton = require('./lib/DropdownButton');

Object.defineProperty(exports, 'DropdownButton', {
  enumerable: true,
  get: function () {
    return _interopRequireDefault(_DropdownButton).default;
  }
});

var _TextButton = require('./lib/TextButton');

Object.defineProperty(exports, 'TextButton', {
  enumerable: true,
  get: function () {
    return _interopRequireDefault(_TextButton).default;
  }
});

var _IconButtonToggleGroup = require('./lib/IconButtonToggleGroup');

Object.defineProperty(exports, 'IconButtonToggleGroup', {
  enumerable: true,
  get: function () {
    return _interopRequireDefault(_IconButtonToggleGroup).default;
  }
});

var _types = require('./lib/types');

Object.keys(_types).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _types[key];
    }
  });
});

var _Button2 = _interopRequireDefault(_Button);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = _Button2.default;
