// @workday/canvas-kit-react-core/dist/commonjs/lib/type.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
var canvas_colors_web_1 = __importStar(require("@workday/canvas-colors-web"));
exports.fontFamily = '"Roboto", "Helvetica Neue", "Helvetica", Arial, sans-serif';
exports.monoFontFamily = '"Roboto Mono", "Courier New", Courier, monospace';
var hierarchy = {
    dataViz1: {
        fontSize: '56px',
        lineHeight: '68px',
        fontWeight: 300,
        color: canvas_colors_web_1.typeColors.heading,
    },
    dataViz2: {
        fontSize: '34px',
        lineHeight: '40px',
        fontWeight: 300,
        color: canvas_colors_web_1.typeColors.heading,
    },
    h1: {
        fontSize: '28px',
        lineHeight: '36px',
        fontWeight: 500,
        color: canvas_colors_web_1.typeColors.heading,
    },
    h2: {
        fontSize: '24px',
        lineHeight: '32px',
        fontWeight: 700,
        color: canvas_colors_web_1.typeColors.heading,
    },
    h3: {
        fontSize: '20px',
        lineHeight: '28px',
        fontWeight: 700,
        color: canvas_colors_web_1.typeColors.heading,
    },
    h4: {
        fontSize: '16px',
        lineHeight: '24px',
        fontWeight: 700,
        color: canvas_colors_web_1.typeColors.heading,
    },
    h5: {
        fontSize: '16px',
        lineHeight: '24px',
        fontWeight: 400,
        color: canvas_colors_web_1.typeColors.heading,
    },
    body: {
        fontSize: '14px',
        lineHeight: '20px',
        fontWeight: 400,
        color: canvas_colors_web_1.typeColors.body,
    },
    body2: {
        fontSize: '13px',
        lineHeight: '20px',
        fontWeight: 400,
        color: canvas_colors_web_1.typeColors.body,
    },
    small: {
        fontSize: '12px',
        lineHeight: '16px',
        fontWeight: 400,
        color: canvas_colors_web_1.typeColors.body,
    },
};
Object.keys(hierarchy).forEach(function (key) {
    hierarchy[key] = __assign({}, hierarchy[key], { fontFamily: exports.fontFamily });
});
var variants = {
    label: {
        fontWeight: 500,
    },
    button: {
        fontWeight: 500,
    },
    caps: {
        fontWeight: 700,
        textTransform: 'uppercase',
    },
    hint: {
        color: canvas_colors_web_1.typeColors.hint,
    },
    error: {
        color: canvas_colors_web_1.statusColors.error,
    },
    inverse: {
        color: canvas_colors_web_1.typeColors.inverse,
    },
    mono: {
        fontFamily: exports.monoFontFamily,
    },
    link: {
        textDecoration: 'none',
        color: canvas_colors_web_1.typeColors.link,
        cursor: 'pointer',
        '&:hover, &:active': {
            textDecoration: 'underline',
            color: canvas_colors_web_1.typeColors.link,
        },
        '&:focus': {
            background: canvas_colors_web_1.default.blueberry100,
            textDecoration: 'underline',
            outline: "2px solid " + canvas_colors_web_1.default.blueberry100,
        },
    },
};
var type = __assign({}, hierarchy, { variant: __assign({}, variants) });
exports.default = type;
